test/timeout.bats

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	helptext=$(crictl runp -h)
+	if [[ "$helptext" != *"cancel-timeout"* ]]; then
+		skip "must have a crictl with the -T option to test CRI-O's timeout handling"
+	fi
+	setup_test
+}
+
+function teardown() {
+	cleanup_test
+}
+
+function create_conmon() {
+	timeout=$1
+	cat > "$TESTDIR"/tmp_conmon << EOF
+#!/bin/sh
+if [[ "\$1" != "--version" ]]; then
+	sleep $timeout
+fi
+$CONMON_BINARY \$@
+EOF
+	chmod +x "$TESTDIR/tmp_conmon"
+}
+
+@test "should not clean up pod after timeout" {
+	create_conmon 2s
+	# need infra container so we can timeout in conmon
+	CONTAINER_DROP_INFRA=false CONTAINER_CONMON="$TESTDIR/tmp_conmon" start_crio
+	run crictl runp -T 2s "$TESTDATA"/sandbox_config.json
+	echo "$output"
+	[[ "$output" == *"context deadline exceeded"* ]]
+	[ "$status" -ne 0 ]
+
+	# allow cri-o to catch up
+	sleep 4s
+
+	# cri-o should not report any pods
+	pods=$(crictl pods -q)
+	[[ -z "$pods" ]]
+
+	created_ctr_id=$("$CONTAINER_RUNTIME" --root "$RUNTIME_ROOT" list -q)
+	[[ ! -z "$created_ctr_id" ]]
+
+	run crictl runp "$TESTDATA"/sandbox_config.json
+	echo "$output"
+	[ "$status" -eq 0 ]
```

Comment:
```suggestion
	output=$(crictl runp "$TESTDATA"/sandbox_config.json)
```

In this case
 - stderr (in which we might be interested it if the test is failing) is not redirected (and will be shown by bats if test fails);
 - stdout (which we're not really interested in except for the below check) goes to `$output`;
 - exit code check is implicit.